### PR TITLE
Add service account name in credential request

### PR DESCRIPTION
**Pull request by akihikokuroda**
The name of the service account to be patched can be specified in the create credential request.

### CODE REVIEW

**Comment by akihikokuroda** [comment]
This PR is for the issue https://github.com/tektoncd/dashboard/issues/76

File: pkg/endpoints/credentials_test.go [hunk 1]
```diff
@@ -63,7 +65,7 @@ func TestCredentials(t *testing.T) {
 	// CREATE credential accesstoken
 	t.Log("CREATE credential accesstoken")
 	createCredentialTest(namespace, accessTokenCred, "", r, t)
-	if !r.checkSecretInSa(namespace, "default", "credentialaccesstoken", t) {
+	if !r.checkSecretInSa(namespace, "sa1", "credentialaccesstoken", t) {
```
Comment (mnuttall) [inline pkg/endpoints/credentials_test.go]: This looks like it's checking that accessTokenCred is to be patched onto sa1. Is that right? Should there also be a test that credentialuserpass is patched onto the default service account?

File: pkg/endpoints/credentials.go [hunk 2]
```diff
@@ -334,6 +335,16 @@ func getQueryEntity(entityPointer interface{}, request *restful.Request, respons
 
 // Convert K8s secret struct into credential struct
 func secretToCredential(secret *corev1.Secret, mask bool) credential {
+	labels := secret.GetLabels()
+	saName, ok := labels[serviceAccountLabelKey]
+	if !ok {
```
[the remark this answers is not in the document]
Comment (akihikokuroda) [reply pkg/endpoints/credentials.go]: OK.  I delete the lines.

File: pkg/endpoints/credentials.go [hunk 3]
```diff
@@ -362,9 +374,21 @@ func credentialToSecret(cred credential, namespace string, response *restful.Res
...
+	if cred.ServiceAccount != "" {
+		saName = cred.ServiceAccount
+	} else {
+		providedSaName := os.Getenv("PIPELINE_RUN_SERVICE_ACCOUNT")
+		if providedSaName != "" {
+			saName = providedSaName
+		} else {
+			saName = "default"
+		}
+	}
```
[the remark this answers is not in the document]
Comment (akihikokuroda) [reply pkg/endpoints/credentials.go]: OK.  I remove the second return value.

File: pkg/endpoints/credentials_test.go (hunk 1 above)
Comment (akihikokuroda) [reply pkg/endpoints/credentials_test.go]: Yes, it is tested around line 75.

**Pull request merged**